### ISSUE TITLE
测试结果？

### ISSUE THREAD

**Issue by cjjjy** [bamtercelboo/pytorch_NER_BiLSTM_CNN_CRF#9]
你好，我利用main.py训练了MSRA语料example.trai，最终生成中文 NER模型，然后运行run_test.sh识别example.test测试集，最终生成结果example.test.shuffle.out

运行main.py，训练过程中precision，recall,f-score,TAG-ACC都为0
## The 1 Epoch, All 1000 Epochs ! ##
now lr is 0.015
Train Time 1.252
Dev eval: precision = 0.000000%  recall = 0.000000% , f-score = 0.000000%,  [TAG-ACC = 0.000000%]
Dev Time 0.148
Test eval: precision = 0.000000%  recall = 0.000000% , f-score = 0.000000%,  [TAG-ACC = 0.000000%]
The Current Best Dev F-score: 0.000000, Locate on 1 Epoch.
The Current Best Test Result: precision = 0.000000%  recall = 0.000000% , f-score = 0.000000%
Test Time 0.164
save best model to ./Save_BModel/ner_model.pt

## The 2 Epoch, All 1000 Epochs ! ##
now lr is 0.014285714285714285
Train Time 1.231
Dev eval: precision = 0.000000%  recall = 0.000000% , f-score = 0.000000%,  [TAG-ACC = 0.000000%]
Dev Time 0.160
Test eval: precision = 0.000000%  recall = 0.000000% , f-score = 0.000000%,  [TAG-ACC = 0.000000%]
The Current Best Dev F-score: 0.000000, Locate on 2 Epoch.
The Current Best Test Result: precision = 0.000000%  recall = 0.000000% , f-score = 0.000000%
Test Time 0.159
save best model to ./Save_BModel/ner_model.pt

## The 3 Epoch, All 1000 Epochs ! ##
now lr is 0.013636363636363634
Train Time 1.190
Dev eval: precision = 0.000000%  recall = 0.000000% , f-score = 0.000000%,  [TAG-ACC = 0.000000%]
Dev Time 0.154
Test eval: precision = 0.000000%  recall = 0.000000% , f-score = 0.000000%,  [TAG-ACC = 0.000000%]
The Current Best Dev F-score: 0.000000, Locate on 3 Epoch.
The Current Best Test Result: precision = 0.000000%  recall = 0.000000% , f-score = 0.000000%
Test Time 0.161
save best model to ./Save_BModel/ner_model.pt

## The 4 Epoch, All 1000 Epochs ! ##
now lr is 0.013043478260869566
Train Time 1.197
Dev eval: precision = 0.000000%  recall = 0.000000% , f-score = 0.000000%,  [TAG-ACC = 0.000000%]
Dev Time 0.157
Test eval: precision = 0.000000%  recall = 0.000000% , f-score = 0.000000%,  [TAG-ACC = 0.000000%]
The Current Best Dev F-score: 0.000000, Locate on 4 Epoch.
The Current Best Test Result: precision = 0.000000%  recall = 0.000000% , f-score = 0.000000%
Test Time 0.160
save best model to ./Save_BModel/ner_model.pt
。。。。。。。。
。。
。。
。。
## The 15 Epoch, All 1000 Epochs ! ##
now lr is 0.008823529411764704
Train Time 1.243
Dev eval: precision = 0.000000%  recall = 0.000000% , f-score = 0.000000%,  [TAG-ACC = 0.000000%]
Dev Time 0.163
Test eval: precision = 0.000000%  recall = 0.000000% , f-score = 0.000000%,  [TAG-ACC = 0.000000%]
The Current Best Dev F-score: 0.129702, Locate on 5 Epoch.
The Current Best Test Result: precision = 0.000000%  recall = 0.000000% , f-score = 0.000000%
Test Time 0.161
Dev Has Not Promote 10 / 10
Early Stop Train. Best Score Locate on 5 Epoch.
#===================================

example.test.shuffle.out结果如下：
沏 O B-PER
茶 O B-PER
时 O B-PER
， O B-PER
一 O B-PER
杯 O B-PER
清 O B-PER
水 O B-PER
被 O B-PER
新 O B-PER
茶 O B-PER
的 O B-PER
细 O B-PER
芽 O B-PER
嫩 O B-PER
叶 O B-PER
染 O B-PER
绿 O B-PER
， O B-PER
春 O B-PER
色 O B-PER
满 O B-PER
杯 O B-PER
， O B-PER
清 O B-PER
香 O B-PER
幽 O B-PER
远 O B-PER
。 O B-PER

仅 O B-PER
仅 O B-PER
迎 O B-PER
来 O B-PER
送 O B-PER
往 O B-PER
， O B-PER
那 O B-PER
礼 O B-PER
节 O B-PER
就 O B-PER
多 O B-PER
得 O B-PER
不 O B-PER
得 O B-PER
了 O B-PER
。 O B-PER

为 O B-PER
了 O B-PER

如果第二列是test测试集正确标签,第三列是预测标签，感觉出入有点大
 =============================  
config.cfg如下
[Embed]
pretrained_embed = False
zeros = False
avg = False
uniform = False
nnembed = True
pretrained_embed_file = 
[Data]
train_file = ./Data/MSRA/example.trai
dev_file = ./Data/MSRA/example.dev
test_file = ./Data/MSRA/example.test
max_count = -1
min_freq = 1
shuffle = True
epochs_shuffle = True

[Save]
save_pkl = True
pkl_directory = ./Save_pkl
pkl_data = pkl_data.pkl
pkl_alphabet = pkl_alphabet.pkl
pkl_iter = pkl_iter.pkl
pkl_embed = pkl_embed.pkl
save_dict = True
dict_directory = ./Save_dictionary
word_dict = dictionary_word.txt
label_dict = dictionary_label.txt
save_direction = ./Save_model
save_best_model_dir = ./Save_BModel
save_model = True
save_all_model = False
save_best_model = True
model_name = ner_model
rm_model = True

[Model]
average_batch = False
use_crf = True
use_char = True
model_bilstm = True
model_bilstm_context = False
lstm_layers = 1
embed_dim = 100
embed_finetune = True
lstm_hiddens = 200
dropout_emb = 0.5
dropout = 0.5
max_char_len = 20
char_dim = 30
conv_filter_sizes = 3
conv_filter_nums = 30
windows_size = 5

[Optimizer]
adam = False
sgd = True
learning_rate = 0.015
weight_decay = 1.0e-8
momentum = 0.0
clip_max_norm_use = False
clip_max_norm = None
use_lr_decay = True
lr_rate_decay = 0.05
min_lrate = 0.000005
max_patience = 1

[Train]
num_threads = 1
epochs = 1000
early_max_patience = 10
backward_batch_size = 1
batch_size = 10
dev_batch_size = 10
test_batch_size = 10
log_interval = 100

**Comment by bamtercelboo** [comment]
你使用的MSRA数据是全部数据集嘛？

**Comment by cjjjy** [comment]
不是MSRA全部数据集，训练数据用的代码中的pytorch_NER_BiLSTM_CNN_CRF/Data/MSRA/example.trai

**Comment by bamtercelboo** [comment]
@cjjjy pytorch_NER_BiLSTM_CNN_CRF/Data/MSRA/example.trai, 这里面的数据仅仅是几句样例数据，结果都为0正常，而且对于中文NER数据，和英文NER数据的训练参数配置是不一样的，要重新调整，我建议采用Adam，以及调整学习率等参数。

**Comment by cjjjy** [comment]
谢谢，我还有个问题，就是我有个标好的训练语料文本大约60M左右，但是加载数据和训练时速度很慢。我的显卡配置是2块GTX1080ti

**Comment by cjjjy** [comment]
语料的train sentence 1064042, dev sentence 18318, test sentence 18318.

**Comment by bamtercelboo** [comment]
@cjjjy 中文预料吧，可能在我写的时候没有考虑到大型预料的问题，代码有一些地方是可以速度优化的，你可以进行优化。

**Comment by cjjjy** [comment]
好的。。。

**Comment by bamtercelboo** [comment]
@cjjjy 问一下，60M的数据你现在加载数据的时间是多少，log里面有写
